### PR TITLE
Teardown logic shared for demo/sandbox cmds 

### PR DESCRIPTION
**Pull request by dav009**
# TL;DR
Logic for tearing down sandbox and demo is redundant. This PR unifies teardown logic under a pkg/sandbox package

## Type
- [ ] Bug Fix
- [x] Feature
- [ ] Plugin

## Are all requirements met?

- [ ] Code completed
- [ ] Smoke tested
- [x] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
Unifying repeated logic to `pkg/sandbox`. The groundwork was already done for the start cmd.
- Ran unitests. 
- Created a sandbox and demo clusters then tore them down successfully

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2472

## Follow-up issue
_NA_

### CODE REVIEW

**Review by evalsocket — DISMISSED** [review]
Looks Good, Can you add a unit test for the teardown package?

**Review by pmahindrakar-oss — DISMISSED** [review]

**Review by evalsocket — DISMISSED** [review]

**Review by pmahindrakar-oss — APPROVED** [review]
